Common/src/geometry/CPhysicalGeometry.cpp

```diff
@@ -5218,57 +5182,70 @@ void CPhysicalGeometry::SetPositive_ZArea(CConfig *config) {
 
 void CPhysicalGeometry::SetPoint_Connectivity(void) {
 
+  vector<vector<unsigned long> > points(nPoint);
+
+  SU2_OMP_PARALLEL
+  {
   unsigned short Node_Neighbor, iNode, iNeighbor;
   unsigned long jElem, Point_Neighbor, iPoint, iElem;
 
   /*--- Loop over all the elements ---*/
+  SU2_OMP_MASTER
+  {
+    vector<vector<long> > elems(nPoint);
 
-  for (iElem = 0; iElem < nElem; iElem++)
-
-  /*--- Loop over all the nodes of an element ---*/
-
-    for (iNode = 0; iNode < elem[iElem]->GetnNodes(); iNode++) {
-      iPoint = elem[iElem]->GetNode(iNode);
-
-      /*--- Store the element into the point ---*/
+    for (iElem = 0; iElem < nElem; iElem++) {
 
-      node[iPoint]->SetElem(iElem);
+      /*--- Loop over all the nodes of an element ---*/
+      for (iNode = 0; iNode < elem[iElem]->GetnNodes(); iNode++) {
+        iPoint = elem[iElem]->GetNode(iNode);
+        elems[iPoint].push_back(iElem);
+      }
     }
+    nodes->SetElems(elems);
+  }
+  SU2_OMP_BARRIER
 
   /*--- Loop over all the points ---*/
 
-  for (iPoint = 0; iPoint < nPoint; iPoint++)
+  SU2_OMP_FOR_DYN(roundUpDiv(nPoint,2*omp_get_max_threads()))
+  for (iPoint = 0; iPoint < nPoint; iPoint++) {
 
-  /*--- Loop over all elements shared by the point ---*/
+    /*--- Loop over all elements shared by the point ---*/
 
-    for (iElem = 0; iElem < node[iPoint]->GetnElem(); iElem++) {
+    for (iElem = 0; iElem < nodes->GetnElem(iPoint); iElem++) {
 
-      jElem = node[iPoint]->GetElem(iElem);
+      jElem = nodes->GetElem(iPoint, iElem);
 
       /*--- If we find the point iPoint in the surronding element ---*/
 
-      for (iNode = 0; iNode < elem[jElem]->GetnNodes(); iNode++)
+      for (iNode = 0; iNode < elem[jElem]->GetnNodes(); iNode++) {
 
-        if (elem[jElem]->GetNode(iNode) == iPoint)
+        if (elem[jElem]->GetNode(iNode) != iPoint) continue;
 
         /*--- Localize the local index of the neighbor of iPoint in the element ---*/
 
-          for (iNeighbor = 0; iNeighbor < elem[jElem]->GetnNeighbor_Nodes(iNode); iNeighbor++) {
-            Node_Neighbor = elem[jElem]->GetNeighbor_Nodes(iNode, iNeighbor);
-            Point_Neighbor = elem[jElem]->GetNode(Node_Neighbor);
+        for (iNeighbor = 0; iNeighbor < elem[jElem]->GetnNeighbor_Nodes(iNode); iNeighbor++) {
+          Node_Neighbor = elem[jElem]->GetNeighbor_Nodes(iNode, iNeighbor);
+          Point_Neighbor = elem[jElem]->GetNode(Node_Neighbor);
 
-            /*--- Store the point into the point ---*/
-
-            node[iPoint]->SetPoint(Point_Neighbor);
-          }
+          /*--- Store the point into the point, if it is new ---*/
+          auto End = points[iPoint].end();
+          if (find(points[iPoint].begin(), End, Point_Neighbor) == End)
+            points[iPoint].push_back(Point_Neighbor);
+        }
+      }
     }
+    //sort(points[iPoint].begin(), points[iPoint].end());
```

Comment:
Significant change 1c: Ideally we would sort the neighbor points, but I think that causes a lot of residual changes... so... rain check

Reply:
@economon, based on your comment about deterministic behavior in #963, do you know it the point/edge adjacency has any predefined order at the moment? Is is sorted by global index maybe?

Reply:
No, I do not believe that we sort the neighboring points internally at each node. It will be a function of the mesh (dictated by the ordering in the mesh file), so I am not surprised that the residuals change if you try to sort it.

A tangentially related topic is the RCM renumbering that we do, which requires calling SetPoint_Connectivity() a second time after the renumbering to rebuild the lists. You have probably seen this. Not directly related to the sorting you propose, just another thing to be aware of while changing the point storage.

Reply:
Hmmm good point, maybe doing the sorting only after the RCM would give the benefits without changing all the residuals.

Reply:
I've given up on the sorting idea.